### DISCUSSION TITLE
Capacitor Camera Plugin Questions

### DISCUSSION THREAD

**Discussion by gopherrdiggs** [ionic-team/capacitor#4977] [Q&A]
I was able to get the camera plugin working with my PWA pretty easily, but there were two minor obstacles that I need help understanding:
* I want to prompt the user to either take a photo or select from photos library, so I set the `source` as `CameraSource.Prompt`; however, if the user selects "Take Picture" on an Android or iOS device they see the PWA Elements implementation instead of the native one. To get to the native camera implementation, the user must select "From Photos" and _then_ "Camera," which is not intuitive. The documentation is pretty clear that PWA Elements is required in order for the camera plugin to work, and I found this to be true, but is there a way to configure the plugins to go directly to the native implementation for the "Take Picture" option?
* I wanted to be proactive with the permissions checking in my code, but I found that calling `checkPermissions()` on iOS resulted in the error `Permissions API is not available in this browser.` Removing that line and going straight to `getPhoto()` seems to work, but am I missing something in my installation that would cause that error on iOS?

```
//BUG: This line throws an error on iOS devices
// let perms = await Camera.checkPermissions();

const image = await Camera.getPhoto({
        quality: 50,
        allowEditing: false,
        saveToGallery: true,
        source: CameraSource.Prompt,
        resultType: CameraResultType.DataUrl
      });
```

package.json:
```
"@capacitor/camera": "^1.0.4",
"@capacitor/cli": "^3.2.0",
"@capacitor/core": "^3.2.0",
"@capacitor/ios": "^3.2.0",
"@ionic/pwa-elements": "^3.0.2"
```

**Comment by gopherrdiggs** [comment]
I continued reading through other discussions and found https://github.com/ionic-team/capacitor/discussions/4944 which _seems_ to answer the questions about why `checkPermissions()` throws an error? Not sure if it's the same thing.

**Comment by jcesarmobile** [comment]
Looks like you are running the apps as a PWA and not as native apps, or you are using the web implementation of the plugins.
Can you provide more details and a sample app if possible?

If you are in a PWA, then PWA elements would be used, you can only get the native UI when building native apps.
